pyproj/_crs.pyx

```diff
@@ -2854,24 +2858,16 @@ cdef class _CRS(Base):
             )
             if proj_list != NULL:
                 num_proj_objects = proj_list_get_count(proj_list)
-            if c_out_confidence_list != NULL and num_proj_objects > 0:
```

Comment:
Please don't move this.

Reply:
It is best to cleanup objects as soon as they are no longer needed.

Reply:
I can split it into two try/finally blocks again, but then the error handling is a bit more complex and the code is harder to read. But whatever you prefer is fine.